### PR TITLE
[Bug 1762430] Use ExternalTaskMarker and ExternalTaskSensor in favour of ExternalTaskCompletedSensor

### PR DESCRIPTION
**Pull request by scholtzan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1762430

This PR replaces all usages of `ExternalTaskCompletedSensor` with `ExternalTaskSensor` and adds `ExternalTaskMarker`s to define downstream dependencies.

### CODE REVIEW

File: dags/copy_deduplicate.py [hunk 1]
```diff
@@ -87,6 +87,30 @@
         resources=resources,
     )
 
+    with TaskGroup('copy_deduplicate_all_downstream_external') as copy_deduplicate_all_downstream_external:
+        ExternalTaskMarker(
+            task_id="bhr_collection__wait_for_bhr_ping",
+            external_dag_id="bhr_collection",
+            external_task_id="wait_for_bhr_ping",
+            execution_date="{{ (execution_date + macros.timedelta(hours=4)).isoformat() }}",
```
[the remark this answers is not in the document]
Comment (sean-rose) [reply dags/copy_deduplicate.py:95]: A couple possibilities:
```suggestion
            execution_date="{{ execution_date.replace(hour=5, minute=0).isoformat() }}",
```
```suggestion
            execution_date="{{ ds }}T05:00:00+00:00",
```

File: dags/copy_deduplicate.py [hunk 2]
```diff
@@ -87,6 +87,30 @@
         resources=resources,
     )
 
+    with TaskGroup('copy_deduplicate_all_downstream_external') as copy_deduplicate_all_downstream_external:
```
Comment (sean-rose) [inline dags/copy_deduplicate.py:90]: I like that the `_downstream_external` suffix is precise/unambiguous, but it is a bit lengthy, and in the default tree view that will push the task-specific part of the task ID further behind the run results grid where it'll get more hidden/cut-off.

Maybe just `_downstream` _or_ `_external` would still be clear enough, while being more succinct?

File: dags/glam_fenix.py [hunk 3]
```diff
@@ -67,153 +71,164 @@
...
+        external_dag_id="copy_deduplicate",
+        external_task_id="copy_deduplicate_all",
+        execution_delta=timedelta(hours=1),
+        check_existence=True,
+        mode="reschedule",
+        allowed_states=ALLOWED_STATES,
+        failed_states=FAILED_STATES,
+        pool="DATA_ENG_EXTERNALTASKSENSOR",
+        email_on_retry=False,
         dag=dag,
```
Comment (sean-rose) [inline dags/glam_fenix.py:93]: Do we still need to pass in the DAG via the `dag` parameters everywhere?

I thought one reason for using the context manager approach is that the DAG then automatically gets applied to everything in that context scope and doesn't need to be passed in.

**Review by sean-rose — APPROVED** [review]
r+wc

File: dags/copy_deduplicate.py (hunk 2 above)
Comment (scholtzan) [reply dags/copy_deduplicate.py:90]: `_external` is probably good enough.

**Pull request merged**